### PR TITLE
ADM-438 [Report][frontend] Report response data mapping

### PR DESCRIPTION
**Pull request by fengxinhou**
## Summary

Convert the backend response data into the content UI display
Include
- Deployment frequency
- LeadTime for changes 
- Change failure rate
### Report page
report

## Before

**Screenshots**
Screen Shot 2023-04-13 at 2 36 48 PM

## After
Screen Shot 2023-04-13 at 2 35 34 PM

### CODE REVIEW

File: frontend/src/constants.ts [hunk 1]
```diff
@@ -115,19 +115,20 @@ export enum VelocityMetricsName {
...
-  REVIEW_PROPORTION = 'Total Review Time/Total Cycle Time',
-  TESTING_PROPORTION = 'Total Testing Time/Total Cycle Time',
-  AVERAGE_DEVELOPMENT_TIME = 'Average Development Time',
-  AVERAGE_WAITING_TIME = 'Average Waiting Time',
-  AVERAGE_BLOCK_TIME = 'Average Block Time',
-  AVERAGE_REVIEW_TIME = 'Average Review Time',
-  AVERAGE_TESTING_TIME = 'Average Testing Time',
+  AVERAGE_CYCLE_TIME = 'Average cycle time',
+  DEVELOPMENT_PROPORTION = 'Total development time / Total cycle time',
+  WAITING_PROPORTION = 'Total waiting time / Total cycle time',
```
Comment (spadio9587) [inline frontend/src/constants.ts:120]: same with 'Total waiting for testing time'

File: frontend/src/constants.ts [hunk 2]
```diff
@@ -115,19 +115,20 @@ export enum VelocityMetricsName {
...
+  BLOCK_PROPORTION = 'Total block time / Total cycle time',
+  REVIEW_PROPORTION = 'Total review time / Total cycle time',
+  TESTING_PROPORTION = 'Total testing time / Total cycle time',
+  AVERAGE_DEVELOPMENT_TIME = 'Average development time',
+  AVERAGE_WAITING_TIME = 'Average waiting time',
```
Comment (spadio9587) [inline frontend/src/constants.ts:125]: same with 'Average waiting for testing time'

File: frontend/src/constants.ts [hunk 3]
```diff
@@ -148,3 +149,35 @@ export const INIT_CYCLETIME_METRICS: { id: number; name: string; value: string[]
...
+
+export enum CycleTimeMetrics {
+  AVERAGE_CYCLE_TIME = 'Average Cycle Time',
+  TOTAL_DEVELOPMENT_TIME_DIV_TOTAL_CYCLE_TIME = 'Total Development Time/Total Cycle Time',
+  TOTAL_WAITING_TIME_DIV_TOTAL_CYCLE_TIME = 'Total Waiting Time/Total Cycle Time',
+  TOTAL_BLOCK_TIME_DIV_TOTAL_CYCLE_TIME = 'Total Block Time/Total Cycle Time',
+  TOTAL_REVIEW_TIME_DIV_TOTAL_CYCLE_TIME = 'Total Review Time/Total Cycle Time',
+  TOTAL_TESTING_TIME_DIV_TOTAL_CYCLE_TIME = 'Total Testing Time/Total Cycle Time',
+  AVERAGE_DEVELOPMENT_TIME = 'Average Development Time',
+  AVERAGE_WAITING_TIME = 'Average Waiting Time',
```
Comment (spadio9587) [inline frontend/src/constants.ts:161]: same with 'Average waiting for testing time'

**Review by spadio9587 — COMMENTED** [review]
Little suggestion for rename title name for making sense.

File: frontend/src/constants.ts [hunk 4]
```diff
@@ -136,15 +140,17 @@ export enum Unit {
...
-export const INIT_CYCLETIME_METRICS: { id: number; name: string; value: string[] }[] = [
+export const INIT_CYCLE_TIME_METRICS: { id: number; name: string; value: string[] }[] = [
   {
     id: 1,
     name: 'Cycle time',
     value: ['40'],
   },
 ]
+
+export const GET_STEPS_FAILED = 'get steps failed'
```
Comment (guzhongren) [inline frontend/src/constants.ts:156]: Rename `Get_STEPS_FAILED` to `Get_STEPS_FAILED_MESSAGE`

File: frontend/src/constants.ts [hunk 5]
```diff
@@ -136,15 +140,17 @@ export enum Unit {
...
+    value: ['2'],
   },
 ]
 
-export const INIT_CYCLETIME_METRICS: { id: number; name: string; value: string[] }[] = [
+export const INIT_CYCLE_TIME_METRICS: { id: number; name: string; value: string[] }[] = [
   {
     id: 1,
     name: 'Cycle time',
     value: ['40'],
```
Comment (guzhongren) [inline frontend/src/constants.ts:152]: 1. Change this `value` to `values` or `valueList`
2. Change  type of `40` to a `number` not string

**Review by guzhongren — CHANGES_REQUESTED** [review]
Change with the comments

File: frontend/src/constants.ts [hunk 6]
```diff
@@ -136,15 +140,17 @@ export enum Unit {
 export const INIT_VELOCITY_METRICS: { id: number; name: string; value: string[] }[] = [
   {
     id: 1,
-    name: '2',
-    value: ['dhjejh'],
+    name: 'velocity',
+    value: ['2'],
```
[the remark this answers is not in the document]
Comment (fengxinhou) [reply frontend/src/constants.ts:144]: The data returned from the backend is string.

File: frontend/src/hooks/reportMapper/leadTimeForChanges.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    mappedLeadTimeForChangesValue.push(deploymentFrequencyValue)
+  })
+  mappedLeadTimeForChangesValue.push({
+    id: mappedLeadTimeForChangesValue.length,
+    name: `${avgLeadTimeForChanges.name}/`,
+    values: Object.entries(avgLeadTimeForChanges)
+      .slice(-3)
+      .map(([name, value]) => ({
+        name: name,
+        value: value.toString(),
```
[the remark this answers is not in the document]
Comment (fengxinhou) [reply frontend/src/hooks/reportMapper/leadTimeForChanges.ts:30]: Other data types are strings. We use a universal model, so in order to maintain consistency in data types, we convert it to String.

File: frontend/src/constants.ts (hunk 5 above)
Comment (fengxinhou) [reply frontend/src/constants.ts:152]: The data returned from the backend is string.

File: frontend/src/constants.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (guzhongren) [reply frontend/src/constants.ts:144]: Negotiate with backend/mockServer to change it and make the data format sense

**Review by guzhongren — COMMENTED** [review]
Chang in the next PR

File: frontend/src/constants.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (TingLiMio) [reply frontend/src/constants.ts]: Chang in the next PR.

**Review by ZhiqiaoYang — APPROVED** [review]

**Review by spadio9587 — APPROVED** [review]
LGTM

**Review by guzhongren — APPROVED** [review]

**Pull request merged**